### PR TITLE
Custom docker root

### PR DESCRIPTION
**Pull request by aiwantaozi**
Issue: https://github.com/rancher/rancher/issues/13304
Types: https://github.com/rancher/types/pull/505

### CODE REVIEW

File: pkg/controllers/user/logging/clusterlogging.go [hunk 1]
```diff
@@ -298,7 +298,12 @@ func provision(namespaces v1.NamespaceInterface, configmaps v1.ConfigMapInterfac
 		return err
 	}
 
-	return utils.CreateFluentd(daemonsets, serviceAccounts, clusterRoleBindings, loggingconfig.LoggingNamespace, dockerRootDir)
+	cluster, err := clusterLister.Get("", clusterName)
```
Comment (alena1108) [inline pkg/controllers/user/logging/clusterlogging.go]: should we handle not found error specially? for the case when cluster is being removed.

**Comment by alena1108** [comment]
@aiwantaozi @cloudnautique Just a general comment: is it possible for the dockerrootdir being different cross nodes in the same cluster? I see that we already have it as a parameter on the node status for machines we provision: https://github.com/rancher/types/blob/master/apis/management.cattle.io/v3/machine_types.go#L75

and now adding it on the cluster level - looks like it can just shadow the actual state of things? https://github.com/rancher/types/pull/505/files#diff-fb6a90b273f7a0747ca4791359846dbdR70

**Comment by aiwantaozi** [comment]
@alena1108 The docker root is been used by the fluentd daemonset, we need to mount the docker root so we could get the log. Since the fluentd is a daemonset, and don't know the docker root before it been scheduled to a node, so we want user to tell us their docker root, and just support one docker root in each cluster.

File: pkg/controllers/user/logging/clusterlogging.go (hunk 1 above)
Comment (aiwantaozi) [reply pkg/controllers/user/logging/clusterlogging.go]: if the cluster is removed, these controllers will also be removed finally, maybe we don't need to handle the not found error?

**Comment by alena1108** [comment]
The commit got merged as https://github.com/rancher/rancher/pull/14667

**Pull request closed without merging**